src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -78,33 +98,177 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
+                }
+                if (entry.getInbook() != null) {
+                    bibEntry.setType("inbook");
+                    parseInbook(entry.getInbook(), fields);
+                }
+                if (entry.getIncollection() != null) {
+                    bibEntry.setType("incollection");
+                    Incollection incollection = entry.getIncollection();
+                    if (incollection.getChapter() != null) {
+                        fields.put(FieldName.CHAPTER, String.valueOf(incollection.getChapter()));
+                    }
+                    parse(incollection, fields);
+                }
+                if (entry.getInproceedings() != null) {
+                    bibEntry.setType("inproceedings");
+                    parse(entry.getInproceedings(), fields);
+                }
+                if (entry.getManual() != null) {
+                    bibEntry.setType("manual");
+                    parse(entry.getManual(), fields);
+                }
+                if (entry.getMastersthesis() != null) {
+                    bibEntry.setType("mastersthesis");
+                    parse(entry.getMastersthesis(), fields);
+                }
+                if (entry.getMisc() != null) {
+                    bibEntry.setType("misc");
+                    parse(entry.getMisc(), fields);
+                }
+                if (entry.getPhdthesis() != null) {
+                    bibEntry.setType("phdthesis");
+                    parse(entry.getPhdthesis(), fields);
+                }
+                if (entry.getProceedings() != null) {
+                    bibEntry.setType("proceedings");
+                    parse(entry.getProceedings(), fields);
+                }
+                if (entry.getTechreport() != null) {
+                    bibEntry.setType("techreport");
+                    parse(entry.getTechreport(), fields);
+                }
+                if (entry.getUnpublished() != null) {
+                    bibEntry.setType("unpublished");
+                    parse(entry.getUnpublished(), fields);
+                }
+                if (entry.getId() != null) {
+                    bibEntry.setCiteKey(entry.getId());
+                }
+                bibEntry.setField(fields);
+                bibItems.add(bibEntry);
+            }
+        } catch (JAXBException | XMLStreamException e) {
             LOGGER.error("Error with XML parser configuration", e);
             return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (org.xml.sax.SAXException e) {
-            LOGGER.error("Error during XML parsing", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (IOException e) {
-            LOGGER.error("Error during file import", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
         }
         return new ParserResult(bibItems);
     }
 
+    /**
```

Comment:
Add some more explanation why you choose that implementation method and not working on the real classes.